File: java tasks/lab3Task/src/main/java/com/mycompany/lab3task/Rectangle.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mycompany.lab3task;

/**
 *
 * @author Zeyad
 */
public class Rectangle extends Shape{
    private double height;
    
    public Rectangle(double h, double w){
        super(w);
        height = h;
    }

    public double getHeight() {
        return height;
    }

    public void setHeight(double height) {
        this.height = height;
    }
    
    @Override
    public double calculateArea(){
        return dim*height;
    }
}

File: Kotlin/Lap 2.1/src/Rectangle.kt
class Rectangle() : Shape() {
    private var height: Double = 0.0

    constructor(w: Double, h: Double) : this() {
        setDim(w)
        height = h
    }
    fun getHeight(): Double {
        return height
    }
    fun setHeight(h: Double) {
        height = h
    }
    override fun calcArea(): Double {
        return getDim() * height
    }

}
